src/box/xlog.c

```diff
@@ -732,12 +732,19 @@ xdir_collect_inprogress(struct xdir *xdir)
 			continue;
 
 		char path[PATH_MAX];
-		snprintf(path, sizeof(path), "%s/%s", dirname, dent->d_name);
+		int would_be_written = snprintf(path, sizeof(path), "%s/%s",
+						dirname, dent->d_name);
+		if ((size_t)would_be_written >= sizeof(path)) {
+			say_error("in-progress file's path got truncated: %s/%s",
+				  dirname, dent->d_name);
+			goto out_closedir;
```

Comment:
Nit:
- The error message looks confusing: a path was truncated - so what?
- No need to add a label - to break the loop, you can use `break`. Anyway, I don't see much point in breaking the loop.

Suggested diff:

```diff
diff --git a/src/box/xlog.c b/src/box/xlog.c
index cedbdff3bb23..bec53c7d4751 100644
--- a/src/box/xlog.c
+++ b/src/box/xlog.c
@@ -730,10 +730,16 @@ xdir_collect_inprogress(struct xdir *xdir)
 		char *ext = strrchr(dent->d_name, '.');
 		if (ext == NULL || strcmp(ext, inprogress_suffix) != 0)
 			continue;
-
+		int rc;
 		char path[PATH_MAX];
-		snprintf(path, sizeof(path), "%s/%s", dirname, dent->d_name);
-		if (unlink(path) < 0)
+		if ((size_t)snprintf(path, sizeof(path), "%s/%s",
+				     dirname, dent->d_name) >= sizeof(path)) {
+			errno = ENAMETOOLONG;
+			rc = -1;
+		} else {
+			rc = unlink(path);
+		}
+		if (rc != 0)
 			say_syserror("error while removing %s", path);
 		else
 			say_info("removed %s", path);
```

Reply:
I would rather show the expected path in `say_syserror`, i.e.:
```C
say_syserror("error while removing %s/%s", dirname, dent->d_name);
```

Reply:
I also believe mimicing a system error here is kind of wrong: if you look at the POSIX specification of [`unlink`](https://pubs.opengroup.org/onlinepubs/007908799/xsh/unlink.html), it fails with `ENAMETOOLONG` under completely different conditions.

Reply:
I checked the POSIX standard and mimicking a system error here is indeed correct.

Reply:
Applied your diff with my nit above.